File: Zig/0002.zig
const std = @import("std");

fn sum_even_fibonacci(limit: u32) u32 {
    var a: u32 = 1;
    var b: u32 = 2;
    var even_sum: u32 = 0;

    while (a <= limit) {
        if (a % 2 == 0) {
            even_sum += a;
        }
        const next = a + b;
        a = b;
        b = next;
    }

    return even_sum;
}

pub fn main() !void {
    const limit: u32 = 4_000_000;
    const result = sum_even_fibonacci(limit);

    std.debug.print("The sum of the even-valued terms is: {}\n", .{result});
}

File: Ruby/0002.rb
def sum_even_fibonacci(limit)
    a, b = 1, 2
    even_sum = 0
  
    while a <= limit
      if a.even?
        even_sum += a
      end
      a, b = b, a + b
    end
  
    even_sum
  end
  
  # Define the limit (4 million)
  limit = 4000000
  
  # Calculate the sum of even-valued Fibonacci numbers
  result = sum_even_fibonacci(limit)
  
  # Print the result
  puts "The sum of the even-valued terms is: #{result}"
  